src/webview/MessageList.js

```diff
@@ -147,19 +147,26 @@ class MessageList extends Component<Props> {
     this.setupSendUpdateEvents();
   }
 
+  componentWillUnmount() {
+    clearInterval(this.readyRetryInterval);
+  }
+
   handleError = (event: mixed) => {
     console.error(event); // eslint-disable-line
   };
 
+  readyRetryInterval: IntervalID | void;
+
   /**
    * Initiate round-trip handshakes with the WebView, until one succeeds.
    */
   setupSendUpdateEvents = (): void => {
-    const intervalId = setInterval(() => {
+    clearInterval(this.readyRetryInterval);
+    this.readyRetryInterval = setInterval(() => {
       if (!this.sendUpdateEventsIsReady) {
         this.sendUpdateEvents([{ type: 'ready' }]);
       } else {
-        clearInterval(intervalId);
+        clearInterval(this.readyRetryInterval);
```

Comment:
Technically, as written, this code is incorrect: the `setInterval` code could reuse the interval ID once it's been cleared, and the unconditional `clearInterval` call at unmount time might clear a real, active interval elsewhere in the application.

... in practice I'm pretty sure all JS engines use incrementing 32-bit integers for interval IDs, so this will never realistically happen. However, I think it'd still be cleaner (fewer possible world-states; simpler to understand, verify, and extend) to keep the interval-ID-field state the same as the interval state.